### PR TITLE
New blog post - CLI for Microsoft 365 context 

### PR DESCRIPTION
**Pull request by Adam-it**
## Category

- [ ] Content fix
- [X] New article

## Contents of the Pull Request

The aim of this PR is to add a new blog post describing one of the latest features added to the CLI for Microsoft 365 which is context

### CODE REVIEW

File: content/post/cli-how-cool-is-that-context/index.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+It's also good to know that it does not make sense to save options short names in the context as CLI for Microsoft 365 under the hood will match command options with options saved in the context only by long name.
+
+## 📑 Why would I use it? What is the use case?
+
+First of all, CLI for Microsoft 365 context gives you a great opportunity to group all of your options in a single place without having to provide them for each executed command, that's just plain convenience and time-saving. 
+
+Let's now think of a scenario in which we have a script that uses CLI for Microsoft 365 commands which we want to execute in a loop providing different set of parameters (options) for each run. Context seems perfect for that task as we may just provide different `.m365rc.json` with different context for each run and ... that's it 🙂.
```
Comment (Jwaegebaert) [inline content/post/cli-how-cool-is-that-context/index.md:63]:
```suggestion
Let's now think of a scenario in which we have a script that uses CLI for Microsoft 365 commands which we want to execute in a loop providing a different set of parameters (options) for each run. The context seems perfect for that task as we may just provide different `.m365rc.json` with different contexts for each run and ... that's it 🙂.
```

File: content/post/cli-how-cool-is-that-context/index.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+Another use case we may think of is what I like to call 'YOLO mode' 😅, which for sure is highly dangerous and I don't recommend it, but it is a good example of how many possibilities and freedom with have with CLI for Microsoft 365 context. In CLI for Microsoft 365 when a `remove` command is executed the user always has to confirm the operation either by answering yes to the prompt or by adding the `--confirm` option to the command. Well, we may forget all of that if we just execute `m365 context option set --name "confirm" --value "true"`. After that CLI will automatically confirm for us any operation 🤯.
```
Comment (Jwaegebaert) [inline content/post/cli-how-cool-is-that-context/index.md:65]: I'm not quite sure how I feel about this part. Although it shows the strengths of the context usage it's quite a big safeguard we are disabling this way. They have been placed to make sure the user doesn't remove or cause any harm by accident to their environment.

**Review by Jwaegebaert — COMMENTED** [review]
Strong work with the blog @Adam-it! 💪 

I don't have a lot to add, so awesome work with the writing job. The few pointers I've left are some typos I spotted during the reading 😄 
Also one topic I'm not quite sure about but we've discussed it already a bit in Basecamp.

File: content/post/cli-how-cool-is-that-context/index.md (hunk 2 above)
Comment (Adam-it) [reply content/post/cli-how-cool-is-that-context/index.md:65]: let's wait for what will be the outcome of the discussion on basecamp but probably I will have to remove the YOLO mode 😜
thanks @Jwaegebaert for your opinion 👍

**Comment by Adam-it** [comment]
Thanks @Jwaegebaert for your review. As always I agree with all 😉

File: content/post/cli-how-cool-is-that-context/index.md (hunk 2 above)
Comment (Adam-it) [reply content/post/cli-how-cool-is-that-context/index.md]: ok removed the YOLO mode and added a different use case. Thanks @Jwaegebaert for your awesome comments 👍

**Comment by LuiseFreese** [comment]
LOVE it! You will merge this on your own?

**Comment by Adam-it** [comment]
> LOVE it! You will merge this on your own? 

TBH I rather you do it 🙂. It's not a release post and don't want to step in anyone shoes 😉

**Comment by LuiseFreese** [comment]
Happy to help - but you do not step on anyone's toes. Merged, site is building now. Thanks for contributing - you are awesome!

**Pull request merged**